### PR TITLE
purple-signald: init at 0.11.0

### PR DESCRIPTION
**Pull request by hufman**
###### Description of changes

Adding a purple plugin to add Signal support, [libpurple-signald](https://github.com/hoehermann/libpurple-signald). nixpkgs already includes signald, this adds the Pidgin/purple plugin for it.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -5224,6 +5224,12 @@
     githubId = 6074754;
     name = "Hlodver Sigurdsson";
   };
+  hufman = {
+    email = "hufman@gmail.com";
+    github = "hufman";
+    githubId = 1592375;
+    name = "Walter Huf";
+  };
```
Comment (azahi) [inline maintainers/maintainer-list.nix:5232]: This needs to be sorted. Check comments on the top of the file.

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-signald/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+stdenv.mkDerivation rec {
+  pname = "purple-signald";
+  version = "0.11.0";
+
+  src = fetchgit {
+    url = "https://github.com/hoehermann/libpurple-signald.git";
+    rev = "ebd7abf91135d2b8f77752ddaffdd989982e55ec";
+    sha256 = "sha256-2LiHjVRBwdPbfravIVM+gvsh3Gq4bhjtRD6eWAbkWmc=";
+  };
+
```
Comment (azahi) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-signald/default.nix:12]:
```suggestion
```

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-signald/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{ lib, stdenv, fetchgit, fetchFromGitHub, pidgin, glib, json-glib, signald }:
```
Comment (azahi) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-signald/default.nix:1]: Looks like this is unused.
```suggestion
{ lib, stdenv, fetchgit, fetchFromGitHub, pidgin, json-glib, signald }:
```

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-signald/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "purple-signald";
+  version = "0.11.0";
+
+  src = fetchgit {
```
Comment (azahi) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-signald/default.nix:7]: Please use `fetchFromGitHub`.

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-signald/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+  PKG_CONFIG_PURPLE_PLUGINDIR = "${placeholder "out"}/lib/purple-2";
+  PKG_CONFIG_PURPLE_DATADIR = "${placeholder "out"}/share";
+
+  installFlags = [ "DESTDIR=$(out)" ];
+
+  meta = with lib; {
+    homepage = "https://github.com/hoehermann/libpurple-signald";
+    description = "Signald plugin for Pidgin";
+    license = licenses.gpl3;
```
Comment (azahi) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-signald/default.nix:28]: This needs to be changed according to [this](https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206).

**Review by azahi — CHANGES_REQUESTED** [review]

**Comment by azahi** [comment]
You also need to have tow separate commits: one for adding yourself as a maintainer, and one for your package.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (hufman) [reply maintainers/maintainer-list.nix:5232]: I understand this needs to be sorted. Is this the wrong place? I thought this order is correct: hlo .. huf .. hug

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-signald/default.nix (hunk 4 above)
Comment (hufman) [reply pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-signald/default.nix:7]: fetchFromGithub seemed unable to fetch the submodules, when I tried before, but I can try again.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (hufman) [reply maintainers/maintainer-list.nix:5232]: Oh I see, hugoreeves is in the wrong place, I'll fix that.

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -5312,12 +5297,33 @@
     githubId = 39689;
     name = "Hugo Tavares Reis";
   };
+  hufman = {
+    email = "hufman@gmail.com";
+    github = "hufman";
+    githubId = 1592375;
+    name = "Walter Huf";
+  };
```
Comment (azahi) [inline maintainers/maintainer-list.nix:5305]: Apologies, I thought this wasn't sorted. Please don't make changes to locations of other attributes as it creates big diffs that are hard to read. You can revert changes to your previous iteration.

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-signald/default.nix [hunk 7]
```diff
@@ -0,0 +1,33 @@
+{ lib, stdenv, fetchFromGitHub, pidgin, json-glib, signald }:
+
+stdenv.mkDerivation rec {
+  pname = "purple-signald";
+  version = "0.11.0";
+
+  src = fetchFromGitHub {
+    owner = "hoehermann";
+    repo = "libpurple-signald";
+    rev = "ebd7abf91135d2b8f77752ddaffdd989982e55ec";
```
Comment (azahi) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-signald/default.nix:10]:
```suggestion
    rev = "v${version}";
```

**Review by azahi — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 6 above)
Comment (hufman) [reply maintainers/maintainer-list.nix:5305]: In this case it only moves 2 other users to be in the proper order, is that still too big of a diff?

Comment (azahi) [reply maintainers/maintainer-list.nix:5305]: You shouldn't _refactor_ this file, but rather just try to "sort" your entry upon adding it. Given the fact that a lot of people could have open PRs that modify this file, keeping your changes small will make it easier to resolve and avoid merge conflicts. It's the same reason why `all-packages.nix` is in a messy state: I bet no one likes that, but there's not much that could be done that will not break the _GitHub flow_.

Comment (hufman) [reply maintainers/maintainer-list.nix]: Very good!

File: maintainers/maintainer-list.nix [hunk 8]
```diff
@@ -5312,6 +5312,12 @@
     githubId = 39689;
     name = "Hugo Tavares Reis";
   };
+  hufman = {
+￼   email = "hufman@gmail.com";
+￼   github = "hufman";
+￼   githubId = 1592375;
+￼   name = "Walter Huf";
+￼ };
```
Comment (azahi) [inline maintainers/maintainer-list.nix:5320]: Something went horribly wrong here :^)
```suggestion
    email = "hufman@gmail.com";
    github = "hufman";
    githubId = 1592375;
    name = "Walter Huf";
  };
```

**Review by azahi — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 8 above)
Comment (hufman) [reply maintainers/maintainer-list.nix:5320]: Bah, copy-pasted from the wrong GitHub text field, it seems!

**Review by azahi — APPROVED** [review]
I did not test this but it builds and looks okay.

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-signald/default.nix [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  ];
+
+  PKG_CONFIG_PURPLE_PLUGINDIR = "${placeholder "out"}/lib/purple-2";
+  PKG_CONFIG_PURPLE_DATADIR = "${placeholder "out"}/share";
+
+  installFlags = [ "DESTDIR=$(out)" ];
+
+  meta = with lib; {
+    homepage = "https://github.com/hoehermann/libpurple-signald";
+    description = "Signald plugin for Pidgin";
```
Comment (winterqt) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-signald/default.nix:28]: It seems like a lot of our Pidgin / libpurple plugins use a description like this (took this format from `purple-matrix`). Probably would be best to mention both Pidgin and libpurple so people can more easily find this when searching for packages.

```suggestion
    description = "Signal support for Pidgin / libpurple";
```

**Review by winterqt — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-signald/default.nix (hunk 9 above)
Comment (hufman) [reply pkgs/applications/networking/instant-messengers/pidgin-plugins/purple-signald/default.nix]: Thank you for the feedback!

**Review by winterqt — APPROVED** [review]

**Pull request merged**